### DISCUSSION TITLE
Storing NodeRef inside Context

### DISCUSSION THREAD

**Discussion by vitordhers** [leptos-rs/leptos#2669] [Q&A]
Folks, is it possible to store `NodeRef` inside a `Context`?

In order to avoid using `web_sys` library's  ```document().get_element_by_id()``` (as it is usually better practice storing `ref`s instead of directly manipulating `DOM` in frameworks), I wanted to store a `NodeRef` inside a `Context`, but that doesn't seem to be possible due to this error:
`this method takes 1 argument but 0 arguments were supplied` when trying this:
```
<input type="text"
    node_ref=my_ctx2.input_ref
/>
```
Please check [this codebox link](https://codesandbox.io/p/devbox/13-actions-0-5-8xk35v?file=%2Fsrc%2Fmain.rs&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyah0r0000082v6rk6m6s6x0%2522%252C%2522sizes%2522%253A%255B62.0323325635104%252C37.9676674364896%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyah0qzz00032v6rzj5f7xtv%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyah0qzz00052v6rla7jy23m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyah0qzz00072v6rcpyafix1%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyah0qzz00032v6rzj5f7xtv%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyah0qzz00022v6r007ss3mi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A1%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.rs%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clyah0qzz00032v6rzj5f7xtv%2522%252C%2522activeTabId%2522%253A%2522clyah0qzz00022v6r007ss3mi%2522%257D%252C%2522clyah0qzz00072v6rcpyafix1%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyah0qzz00062v6rvoeh6xga%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522trunk%2520serve%2522%252C%2522port%2522%253A8000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clyah0qzz00072v6rcpyafix1%2522%252C%2522activeTabId%2522%253A%2522clyah0qzz00062v6rvoeh6xga%2522%257D%252C%2522clyah0qzz00052v6rla7jy23m%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyah0qzz00042v6rk9vxyw0h%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522trunk%2520serve%2522%257D%255D%252C%2522id%2522%253A%2522clyah0qzz00052v6rla7jy23m%2522%252C%2522activeTabId%2522%253A%2522clyah0qzz00042v6rk9vxyw0h%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) in order to check what's happening.

I can confirm the same error applies to `leptos` v0.6.11.

Is that a bug or am I missing anything really fundamental?
Thanks in advance

**Comment by gbj** [comment]
This is unrelated to the question of storing it in context, per se, and seems to just be a syntactic quirk with the view macro not expecting a dotted field there. The codebox link seems wrong to my maybe, but I was able to reproduce the issue and solve it with a local variable.

```rust
    let my_ctx2 = NodeRefObj {
        input_ref: NodeRef::new(),
    };
    let foo = my_ctx2.input_ref;
    view! {
        <input type="text"
            node_ref=foo
        />
    }
```
